keps/sig-apps/4212-declarative-node-maintenance/README.md

```diff
@@ -0,0 +1,1330 @@
+  We have also seen an interest in issue [kubernetes/kubernetes#83307](https://github.com/kubernetes/kubernetes/issues/83307)
+  for overriding evictions, which led to the addition of the `--disable-eviction` flag to
+  `kubectl drain`. There are other examples of this approach on the web .
+- Kevin Reeuwijk, [How to handle blocking PodDisruptionBudgets on K8s with distributed storage](https://www.spectrocloud.com/blog/how-to-handle-blocking-poddisruptionbudgets-on-kubernetes-with-distributed-storage), June 6, 2022 - a simple
+  shell script example on how to drain the node in a safer way. It does a normal eviction, then
+  looks for a pet application (Rook-Ceph in this case) and does hard delete if it does not see it.
+  This approach is not plagued by the loss of data resiliency, but it does require maintenaning a
+  list of pet applications, which can be prone to mistakes. In the end, the cluster admin has to do
+  a job of the application maintainer.
+- Artur Rodrigues, [Impossible Kubernetes node drains](https://www.artur-rodrigues.com/tech/2023/03/30/impossible-kubectl-drains.html), 30 Mar, 2023 - discusses
+  the problem with node drains and offers a workaround to restart the application without the
+  application owner's consents, but acknowledges that this may be problematic without the knowledge
+  of the application
+- Jack Roper, [How to Delete Pods from a Kubernetes Node with Examples](https://spacelift.io/blog/kubectl-delete-pod), 05 Jul, 2023 - also
+  discusses the problem of blocking PDBs and offers several workarounds. Similar to others also
+  offers a force deletion, but also a less destructive method of scaling up the application.
+  However, this also interferes with application deployment and has to be supported by the
+  application.
+
+To sum up. Some projects solve this by introducing validating admission webhooks. This has a couple
+of disadvantages. The webhooks are not easily discoverable by cluster admins. And they can block
+evictions for other applications if they are misconfigured or misbehave. It is not intended for the
+eviction API to be extensible in this way. The webhook approach is therefore not recommended.
+
+As seen in the experience reports and GitHub issues, some admins solve their problems by simply
+ignoring PDBs which can cause unnecessary disruptions or data loss. Some solve this by playing
+with the application deployment, but have to understand that the application supports this.
+
+### Goals
+- Kubectl drain should not evict and disrupt applications with evacuation capability and instead
+  politely ask them to migrate their pods to another node or to remove them by creation of
+  NodeMaintenance object.
+- Introduce a node maintenance controller that will help controllers like deployment controller
+  to migrate their pods.
+- Deployment controller should use `.spec.strategy.rollingUpdate.maxSurge` to evacuate its pods
+  from a node that is under maintenance.
+
+### Non-Goals
```

Comment:
This section is titled `Non-Goals` and describes a list of things in detail which is very confusing. Can we reduce this to `Changes to how PDB status is calculated`, `Change behavior of DaemonSets`, `Prioritization of pods` etc?

Reply:
goals/non-goals have been updated